mobile/lib/services/backup.service.dart

```diff
@@ -335,6 +341,19 @@ class BackupService {
             }
           }
 
+          final localAsset = localAssets.firstWhereOrNull(
+            (localAsset) => localAsset.localId == asset.localId,
+          );
```

Comment:
Looping over all of the assets seems like a bad idea. I think you want [AssetRepository#getDeviceAssetsById](https://github.com/immich-app/immich/blob/f386b4d3771c57560d5d8f973cfb695433703999/mobile/lib/repositories/asset.repository.dart#L154C30-L154C49) instead?